src/main/java/dev/galacticraft/mod/content/block/machine/SimpleMultiBlockMachineBlock.java

```diff
@@ -51,6 +56,21 @@ protected SimpleMultiBlockMachineBlock(Properties properties, ResourceLocation f
         this.partState = partBlock.defaultBlockState();
     }
 
+    @Override
+    protected void onExplosionHit(BlockState state, Level world, BlockPos pos, Explosion explosion, BiConsumer<ItemStack, BlockPos> stackMerger) {
+        if (state.getBlock() != GCBlocks.ADVANCED_SOLAR_PANEL && state.getBlock() != GCBlocks.BASIC_SOLAR_PANEL) {
+            return;
+        }
+        for (BlockPos part : parts) {
+            BlockPos actualPartPos = pos.offset(part);
+            if (!(world.getBlockState(actualPartPos).getBlock() instanceof SolarPanelPartBlock)) {
+                return;
```

Comment:
Prefer `continue` instead of `return` here, as it's possible that one of that parts was already removed (but others weren't)